src/vanilla.ts

```diff
@@ -113,22 +107,17 @@ function createStore<
 
   const destroy: Destroy = () => listeners.clear()
   const api = { setState, getState, subscribe, destroy }
-  state = createState(
-    setState as unknown as CustomSetState,
-    getState as unknown as CustomGetState,
-    api as unknown as CustomStoreApi
+  state = (createState as PopArgument<typeof createState>)(
+    setState,
+    getState,
+    api
   )
-  return api as unknown as CustomStoreApi
+  return api as any
 }
 
-export default createStore
+const createStore = ((f) => {
+  if (f === undefined) return createStoreImpl
+  return createStoreImpl(f)
```

Comment:
This looks a bit different from `react.ts`.
Should we name it `createState` instead of `f` here too?
Also, if we make this ternary syntax, it may reduce bundle size (at least for ESM).
```ts
const createStore = ((createState) => (
  createState ? createImpl(createState) : createImpl
)) as CreateStore
```

Reply:
Yep, renamed and using ternary now.